src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -191,7 +190,11 @@ export class IndexPatternsService {
     return indexPatternCache.set(id, indexPattern);
   };
 
-  specToIndexPattern(spec: IndexPatternSpec) {
+  async specToIndexPattern(spec: IndexPatternSpec) {
+    const shortDotsEnable = await this.config.get(UI_SETTINGS.SHORT_DOTS_ENABLE);
+    const metaFields = await this.config.get(UI_SETTINGS.META_FIELDS);
+    const uiSettingsValues = await this.config.getAll();
```

Comment:
i guess shortDotsEnable and metaFields are also part of getAll ? 
i would prefer not doing getAll at all and rather identify the specific settings that are required if possible ?

Reply:
Currently FieldFormatters can access any uiSetting. getAll is used to support them. We should look at refactoring FieldFormatters so they're not dependent on being passed a dependency like this.